gph/bindings/ripser_bindings.cpp

```diff
@@ -106,4 +132,7 @@ PYBIND11_MODULE(gph_ripser, m)
         "I"_a, "J"_a, "V"_a, "NEdges"_a, "N"_a, "modulus"_a, "dim_max"_a,
         "threshold"_a, "num_threads"_a, "return_generators"_a,
         "ripser sparse distance matrix");
+
+    m.def("get_max_coefficient_field_supported",
+          []() { return (uintptr_t(1) << num_coefficient_bits) - 1; });
```

Comment:
What number is this on Unix and Windows?

Reply:
Which number do you mean ?
Currently because `num_coefficient_bits` is set to 8, it should return 255 on all platforms.

Reply:
Ok! In that case, why not just have a list with all primes up to that value in another file, and import it?   Will save the user computational time and it's not hard to maintain as I doubt we will ever need to offer support for larger primes.

Reply:
The list/set would be

2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 101, 103, 107, 109, 113, 127, 131, 137, 139, 149, 151, 157, 163, 167, 173, 179, 181, 191, 193, 197, 199, 211, 223, 227, 229, 233, 239, 241, 251.

Reply:
The computational time to compute the prime number up to N it not that important (if of course N is not too big).
We could hard-code this, but in my opinion, we should let it flexible.

The reason is that if other user clone the library and they need a bigger number than 251, they will need to update the list. But with current solution, they only need to change the number of bits, and the rest will be done automatically.

Reply:
I profiled this function and it takes about 16 µs to run on my machine.  Computing PH of a point cloud of 30 points takes 1.4 ms, so listing the primes would be about 1% of the runtime, and for 20 points the runtime of PH is around 500 µs (so primes is about 3% of the runtime).

I'm not saying it's a lot, but I think we can do better by stopping early in the computation: e.g.

```
import math

def is_prime(x, N):
    """Test whether x < N is prime."""
    if not x % 2:
        return False

    sieve = [True] * N
    for i in range(3, min(x, int(math.sqrt(N))) + 1, 2):
        if sieve[i]:
            sieve[i * i::2 * i] = [False] * int(((N - i * i - 1) / (2 * i) + 1))

    return sieve[x]
```
will stop almost immediately if the user inputs `3` (which would be the most common use case).

Reply:
It makes sense to change `rwh_primes` to return a boolean to say if the coefficient is prime.
I did not use the method thinking about performances, but if we can avoid spending time unnecessary, why not.

Do you mind if I push your suggestion ?

Reply:
> Do you mind if I push your suggestion ?

Go for it! Beware that it gives `False` when the user passes `2` (but this is treated separately I think?).